jobqueue.go

```diff
@@ -142,9 +205,21 @@ func (s *Stack) newJob() *job {
 // Wait doesn't return other errors than ErrStackFull or ErrTimeout.
 func (s *Stack) Wait() (done func(), err error) {
 	j := s.newJob()
-	s.req <- j
-	err = <-j.notify
-	done = func() { s.done <- token }
+	select {
+	case s.req <- j:
+		err = <-j.notify
+		if err == nil {
```

Comment:
Are you sure that we don't need the done function here?
I remember we had an issue because this.

Reply:
good question. Let me think a bit what would be the safest way. What do you think about returning a noop done function for the case the err != nil?

Reply:
created a noop